lib/plugins/aws/package/compile/events/cognitoUserPool/index.js

```diff
@@ -20,19 +20,21 @@ class AwsCompileCognitoUserPoolEvents {
 
     this.hooks = {
       'package:compileEvents': this.compileCognitoUserPoolEvents.bind(this),
+      'aws:package:finalize:mergeCustomProviderResources': this
```

Comment:
We should update this to use the `after:package:finalize` lifecycle.

The main reason for that is that the lifecycle event which is used here does not ensure order (because we want to merge our Cognito User Pool resources `after` the `merge` has happened) and the `aws:x` lifecycle events are not "official" Serverless lifecycle events.

The Serverless lifecycle events which are not inner lifecycle events are officially marked as non-breaking since they're less likely to be changed.

So in conclusion this should be updated to: `after:package:finalize`